### PR TITLE
first pass at ruff linter fixes

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- fixes many ruff linter errors, removing unused imports and variables
- disables ruff formatting and linting for ipynb files

I verified that all tests pass on a cloud VM.

### CODE REVIEW

**Comment by georgematheos** [comment]
@sritchie thank you for making this PR!

Two questions, from a quick look -- 
1. Have you run the tests successfully on this?  I ask because this removes a few lines of the form `from . import X`, in `Y/__init__.py` files, and I'm wondering if this breaks any of the tests which are expecting to be able to `import Y.X`.
2. It looks like this PR changes a number of lines of the form `varname = expression()` to `expression()`, and for most of these, `expression()` does not have a side effect.  Is the best bet to go through and remove all of these manually?  (Maybe it would be better to try to automatically delete all these lines, and then hope the tests catch if this has broken anything relying on side effects.)

**Comment by sritchie** [comment]
@georgematheos it would definitely be good to go remove those lines, I just didn't know enough myself to know which lines were relying on side effects and which were not. I'm happy to go get it done.

Let me try the tests on this today, and if they look good, what do you think about giving an approval here?

File: b3d/renderer/renderer.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def _rasterize_bwd(self, saved_tensors, diffs):
         )
         dy, ddb = diffs
 
-        grads = _rasterize_bwd_custom_call(
+        _rasterize_bwd_custom_call(
```
Comment (sritchie) [inline b3d/renderer/renderer.py:139]: note to stare at this one more closely

Comment (georgematheos) [reply b3d/renderer/renderer.py:139]: @nishadgothoskar can you please comment on this? -- Do we need to call `_rasterize_bwd_custom_call` here?

Comment (georgematheos) [reply b3d/renderer/renderer.py:139]: (@sritchie I am not familiar with all the code in `renderer.py`, so we probably need Nishad's input here.)

**Pull request merged**